Core/GDCore/Events/Builtin/CommentEvent.cpp

```diff
@@ -12,6 +12,29 @@ using namespace std;
 
 namespace gd {
 
+vector<const gd::String*> CommentEvent::GetAllSearchableStrings( BaseEvent &event)
+    const {
+  vector<const gd::String*> allSearchableStrings;
+
+  stringCom1 = event.GetChild("comment", 0, "Com1").GetValue().GetString();
+  stringCom2 = event.GetChild("comment", 0, "Com2").GetValue().GetString();///< Com2 is deprecated
+  allSearchableStrings.push_back(&stringCom1);
+  allSearchableStrings.push_back(&stringCom2);
+
+  return allSearchableStrings;
+}
+
+vector<gd::String*> CommentEvent::GetAllSearchableStrings( BaseEvent &event) {

```

Comment:
The `const` at the end of the declaration (see the previous function) is here to say "this method won't change the object I swear". If you try to do something that will change the object, the compiler will reject it.

You don't need this second version (called the "non const" version), you can remove it.